### ISSUE TITLE
[Suggestion] Constant Rotation Rotators

### ISSUE THREAD

**Issue by CaptainKronos** [tsunamayo/Starship-EVO#3064]
In essence, setting the angular velocity of a rotator and checking a box that allows it to spin constantly. This would allow people to build fans, rotating lights, and most importantly gravity rings.
This is less of a "hey here's something thats cool here's how it could be implemented" and more of a "are you considering doing this." So, are you considering implementing constant rotation rotators in the future of this game?
It'd be a much appreciated feature, so if it's planned I'm pretty sure we'd all like to know. And if it's not planned, way-hey! Here's a feature people'd like!

**Comment by MatCat** [comment]
I second this, full rotation rotators are a must.

**Comment by tsunamayo** [comment]
Yes I can see why it would be useful, the problem is collisions. That would be a lot of collisions check to do, all the time, potentially completely ruining your fps...
So I have to continue optimizing collisions first and see where we stand.
Cheers

**Comment by AlienXtream** [comment]
you could try precomputing the collision area and just have the rotor not activate at all if something is in the way. justified as a "safety feature" of the rotor lore wise.

**Comment by tsunamayo** [comment]
Hum yeah but the something is in the way part is equal to a collision check of some sort ;) I do a something along those line for turret, that work well as turret are built a certain way. But here people want that for anything, from small light to entire station, so it could not work...

**Comment by CaptainKronos** [comment]
But will you at least implement this in the game at some point? That's all I want to know at the moment.
@tsunamayo

**Comment by tsunamayo** [comment]
@CaptainKronos well I cant commit on that unfortunately. If you need an answer for some reason, then no, it is safer to think that way! Cheers

**Comment by AlienXtream** [comment]
fair enough. im sure you will eventually find a way to make it work. my brain is currently thinking about ways to "fake" it. more theory than practical things. weird stuff that probably isn't even really possible short of programming a rendering engine from scratch with it in mind. the simplest solution to the collision thing though would be to... well... not do collision. it would mean that things like a stanford torus would be impossible (or at least only for show) but would possibly be a way to allow smaller things like propellers or, as you mentioned, lights. not as a permanent solution necessarily but more of an experimental thing to see what people do with it. another option might be to have a setting in the rotor GUI to not rotate back on low signal.

**Comment by Kaiser-Indrasil** [comment]
Well, there is already an infinite rotator on the Steam Workshop. Go check it out!

**Comment by Calvert4096** [comment]
This is brilliant (and also should be unnecessary):
https://steamcommunity.com/sharedfiles/filedetails/?id=2257256842&searchtext=infinite+rotator

**Comment by WILDCATreactor** [comment]
@Calvert4096 woof, so if people are going to build a multi-stage DaVincian rotator, triggering a large number of checks anyways...

**Comment by Calvert4096** [comment]
@tsunamayo  Is there a way you can get the engine to evaluate a "swept volume" when a continuous rotor starts, and just use that until it stops?  Then if a user doesn't want to deal with the consequences of such a "stay out zone", they can use the existing servomotor-like rotor.

**Comment by tsunamayo** [comment]
@Calvert4096 that is just code an new rule for the player to learn, so it is not ideal. That blueprint is just amazing...

**Comment by Uncle-Ulty** [comment]
@tsunamayo, maybe a constant rotation rotator that only checks collision in the first revolution? If it makes the first turn, it will continue to spin. And to avoid future collisions, make it check again if you restart movement (on/off), or if you make any modification add/delete any block on the ship. 

idk, it's just an idea
